src/renderers/shared/ReactDebugTool.js

```diff
@@ -240,15 +249,26 @@ var ReactDebugTool = {
     currentFlushNesting--;
     resumeCurrentLifeCycleTimer();
     emitEvent('onEndFlush');
+    if (isProfiling && hasAdvancedConsoleFeatures && currentFlushNesting === 0) {
+      console.groupEnd('React updates');
+    }
   },
   onBeginLifeCycleTimer(debugID, timerType) {
     checkDebugID(debugID);
     emitEvent('onBeginLifeCycleTimer', debugID, timerType);
+    if (isProfiling && hasAdvancedConsoleFeatures) {
+      var displayName = ReactComponentTreeHook.getDisplayName(debugID);
+      var label = displayName + '#' + timerType;
+      console.time(label);
+    }
     beginLifeCycleTimer(debugID, timerType);
   },
   onEndLifeCycleTimer(debugID, timerType) {
     checkDebugID(debugID);
     endLifeCycleTimer(debugID, timerType);
+    var displayName = ReactComponentTreeHook.getDisplayName(debugID);
+    var label = displayName + '#' + timerType;
+    console.timeEnd(label);
```

Comment:
What happens if you have components that recursively render themselves, would it end an incorrect one? Should we put the debugID in there to make it unique?